5.0/en/0x51-V51-OAuth2.md

```diff
@@ -2,111 +2,86 @@
+| **51.1.4** | [ADDED] Verify that refresh tokens are sender-constrained or use refresh token rotation. | ✓ | ✓ | ✓ |
+| **51.1.5** | [ADDED] Verify that the Authorization Server publishes the element "code_challenge_methods_supported" in their Authorization Server metadata containing the supported PKCE challenge methods. | ✓ | ✓ | ✓ |
 
+## V51.2 OAuth Client
 
-**Client Developer**
 | # | Description | L1 | L2 | L3 |
 | :---: | :--- | :---: | :---: | :---: |
-| **3.8.6** | [ADDED] Verify that when an OAuth Client can interact with more than one Authorization Server, Clients should verify that the issuer "iss" parameter value is what it expected from the authorization response to prevent against mix-up attacks. [reference link for Mix-up attacks: https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics-18#mix_up] In the absence of "iss" parameter, Clients may instead use distinct redirect URIs to identify authorization endpoints and token endpoints.  | ✓ | ✓ | ✓ |
-| **3.8.7** | [ADDED] Verify that the Client is using the PKCE flow or alternatively the OpenID Connect "nonce" parameter and the respective Claim in the ID Token. The PKCE challenge or OpenID Connect "nonce" must be transaction-specific and securely bound to the client and the user agent in which the transaction was started.  | ✓ | ✓ | ✓ |
-| **3.8.8** | [ADDED] Verify that if a Client sends a valid PKCE "code_challenge" parameter in the authorization request, the Authorization Server enforces the correct usage of "code_verifier" at the token endpoint. | ✓ | ✓ | ✓ |
-| **3.8.9** | [ADDED] Verify that Clients are utilizing the parameters "scope" and "resource", respectively to determine the Resource Server they want to access. | ✓ | ✓ | ✓ |
-| **3.8.10** | [ADDED] Verify that Clients are utilizing the parameters "scope" and "authorization_details" to determine the requested/related resources and/or actions the access token are restricted to. | ✓ | ✓ | ✓ |
+| **51.2.1** | [ADDED] Verify that when an OAuth Client can interact with more than one Authorization Server, Clients should verify that the issuer "iss" parameter value is what it expected from the authorization response to prevent against mix-up attacks. [reference link for Mix-up attacks: https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics-18#mix_up] In the absence of "iss" parameter, Clients may instead use distinct redirect URIs to identify authorization endpoints and token endpoints.  | ✓ | ✓ | ✓ |
+| **51.2.2** | [ADDED] Verify that the Client is using the PKCE flow or alternatively the OpenID Connect "nonce" parameter and the respective Claim in the ID Token. The PKCE challenge or OpenID Connect "nonce" must be transaction-specific and securely bound to the client and the user agent in which the transaction was started.  | ✓ | ✓ | ✓ |

```

Comment:
Finalizing my review on this to ensure this nonce's importance/usage is not overlooked. Will push a commit soon. Thanks!

Reply:
Yes because the server needs to create it, not the client. Do you need to add that guidance somewhere else or is that already covered @csfreak92 ?

Reply:
I saw that you added this in the "PKCE" section. Do we not need to mention this somewhere in a requirement @csfreak92 or do we already do so?

Reply:
> I saw that you added this in the "PKCE" section. Do we not need to mention this somewhere in a requirement @csfreak92 or do we already do so?

@tghosth, I moved it to the PKCE section to cover that part about the nonce. I think it does sit well there, don't you think?

Reply:
> Yes because the server needs to create it, not the client. Do you need to add that guidance somewhere else or is that already covered @csfreak92 ?

This one is covered. I moved that nonce comment to the PKCE section. :)